src/thread_pool.cpp

```diff
@@ -35,7 +35,7 @@ thread_pool::~thread_pool()
 
 auto thread_pool::schedule() -> operation
 {
-    if (!m_shutdown_requested.load(std::memory_order::relaxed))
+    if (!m_shutdown_requested.load(std::memory_order::acquire))
```

Comment:
This shouldn't have been `relaxed` since we want to `acquire` any other writes from other threads.